app/models/person.rb

```diff
@@ -425,12 +425,12 @@ def consent
     community_membership.consent
   end
 
-  def is_marketplace_admin?
-    community_membership.admin?
+  def is_marketplace_admin?(community)
+    community_membership.admin? && community_membership.community_id = community.id
```

Comment:
This isn't significant, but to me it would make more sense if the community check is first, before the admin check.